spec/instant.html

```diff
@@ -596,8 +596,12 @@ <h1>
         <dd>It computes the difference between two exact times _ns1_ and _ns2_ expressed in nanoseconds since the epoch, and rounds the result according to the parameters _roundingIncrement_, _smallestUnit_, _largestUnit_, and _roundingMode_.</dd>
       </dl>
       <emu-alg>
-        1. Assert: The following step cannot fail due to overflow in the Number domain because abs(ℝ(_ns2_) - ℝ(_ns1_)) &le; 2 &times; nsMaxInstant.
-        1. Let _roundResult_ be ! RoundDuration(0, 0, 0, 0, 0, 0, 0, 0, 0, ℝ(_ns2_) - ℝ(_ns1_), _roundingIncrement_, _smallestUnit_, _roundingMode_).[[DurationRecord]].
+        1. Let _difference_ be BigInt::subtract(_ns2_, _ns1_).
+        1. Let _nanoseconds_ be ℝ(! BigInt::remainder(_difference_, *1000*<sub>ℤ</sub>)).
+        1. Let _microseconds_ be ℝ(! BigInt::remainder(! BigInt::divide(_difference_, *1000*<sub>ℤ</sub>), *1000*<sub>ℤ</sub>)).
+        1. Let _milliseconds_ be ℝ(! BigInt::remainder(! BigInt::divide(_difference_, *1000000*<sub>ℤ</sub>), *1000*<sub>ℤ</sub>)).
+        1. Let _seconds_ be ℝ(! BigInt::divide(_difference_, *1000000000*<sub>ℤ</sub>)).
```

Comment:
I would suggest performing all the arithmetic in the ℝ domain. It should be the same result, the BigInt operations do the calculations in ℝ anyway, and then truncate. But I think it will be clearer to read in the spec because the reader doesn't have to look at the BigInt operations.